### PR TITLE
refactor: remove obsolete ResultsDecorator types

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

Removes all `ResultsDecorator` references and corresponding types by replacing them with equivalent undecorated axe types. We weren't using the decoration results anyway, and have already used a different strategy in the 21 converter.

#### Pull request checklist

- [x] PR title respects [Conventional Commits](https://www.conventionalcommits.org) (starts with `fix:`, `feat:`, etc, and is suitable for user-facing release notes)
- [x] PR contains no breaking changes, **OR** description of both PR **and final merge commit** starts with `BREAKING CHANGE:`
- [n/a] (if applicable) Addresses issue: #0000
- [x] Added relevant unit tests for your changes
- [x] Ran `yarn precheckin`
- [x] Verified code coverage for the changes made

### CODE REVIEW

File: src/environment-data-provider.test.ts [hunk 1]
```diff
@@ -12,22 +12,23 @@ describe('environment-data-provider', () => {
...
-                targetPageUrl: stubTargetPageUrl,
+            const stubAxeResults: Axe.AxeResults = {
+                url: stubTargetPageUrl,
                 timestamp: stubTimestamp,
-                targetPageTitle: stubTargetPageTitle,
                 passes: [],
                 incomplete: [],
                 violations: [],
                 inapplicable: [],
+                toolOptions: null!,
```
Comment (lisli1) [inline src/environment-data-provider.test.ts]: Just wondering, what is the purpose of the ! operator here? I did a quick search and saw that it is commonly used as a "non-null assertion operator" but it is null here, so I am a bit confused.

File: src/sarif-converter-21.test.ts [hunk 2]
```diff
@@ -87,15 +85,18 @@ describe('SarifConverter21', () => {
...
                 passes: [],
                 violations: [],
                 inapplicable: [],
                 incomplete: [],
-            };
+                toolOptions: null!,
+                testEngine: null!,
+                testRunner: null!,
+                testEnvironment: null!,
+            } as Axe.AxeResults;
```
Comment (lisli1) [inline src/sarif-converter-21.test.ts]: Do we need to cast `stubResults` as `Axe.AxeResults` if we are already asserting that it is of that type (in `const stubResults: Axe.AxeResults`)?

File: src/environment-data-provider.test.ts (hunk 1 above)
Comment (dbjorge) [reply src/environment-data-provider.test.ts]: The `!` forces typescript to believe that the input will be acceptable even for a non-null-accepting type. But I appreciate the feedback that this is confusing; I think the way you did it in your PR (`{} as Axe.ToolOptions`) is less confusing, so I'll switch to that

File: src/sarif-converter-21.test.ts (hunk 2 above)
Comment (dbjorge) [reply src/sarif-converter-21.test.ts]: Oops, this was leftover from an earlier attempt at avoiding the `null!`s

**Comment by dbjorge** [comment]
Closing and reopening to trigger CLA check

**Pull request merged**